bite-size-notes/week-8-c-LQG-stability-smvial.ipynb

```diff
@@ -0,0 +1,363 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 109,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Steady state Sigma:\n",
+      "[[ 5.  5.]\n",
+      " [ 5. 10.]]\n",
+      "Kallman Gain, L:  [5. 5.]\n",
+      "Analytically derived L:  [5. 5.]\n"
+     ]
+    }
+   ],
+   "source": [
+    "C = np.array([1, 0])\n",
+    "sig_w = 5\n",
+    "S_w = q_k * np.ones((2,2))\n",
+    "S_v = np.array([1])\n",
+    "# R_k = r_k * np.ones((2,2))\n",
+    "# Cov_p = sigma_p * np.ones((2,2))\n",
+    "\n",
+    "# Solve the continuous time ARE for \n",
+    "Sigma = sp.linalg.solve_continuous_are(A.T, C.reshape(2,1), S_w, S_v )\n",
+    "print(\"Steady state Sigma:\")\n",
+    "print(Sigma)\n",
+    "\n",
+    "# Solve for Kallman gain L\n",
+    "L = Sigma @ C.reshape(2,1) @ S_v\n",
+    "print(\"Kallman Gain, L: \", L)\n",
+    "\n",
+    "# Check against analytic solution\n",
+    "L_a = (2 + np.sqrt(4+sig_w)) * np.ones(2)\n",
+    "print(\"Analytically derived L: \", L_a)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let us simplify our control and Kallman gains further for the sake of coming notation, and say\n",
+    "$$\n",
+    "\n",
+    "K = f \\begin{bmatrix}\n",
+    "1 & 1 \n",
+    "\\end{bmatrix} \\qquad \\text{ and } \\qquad L = d\\begin{bmatrix}\n",
+    "1 \\\\ \n",
+    "1 \n",
+    "\\end{bmatrix}\n",
+    "\n",
+    "\n",
+    "$$\n",
+    "\n",
+    "\n",
+    "Putting all of this together, our full system dynamics become:\n",
+    "$$\n",
+    "\n",
+    "\\begin{bmatrix}\n",
+    "A & BK\\\\\n",
+    "LC & A-LC-BK \n",
+    "\\end{bmatrix} \n",
+    "\\begin{bmatrix}\n",
+    "x \\\\ \n",
+    "\\hat{x} \n",
+    "\\end{bmatrix} = \n",
+    "\\begin{bmatrix}\n",
+    "\\dot{x} \\\\ \n",
+    "\\dot{\\hat{x}} \n",
+    "\\end{bmatrix}\n",
+    "\n",
+    "$$\n",
+    "\n",
+    "Here $\\hat{x}$ is our extimate of $x$ from observation. This system is comes from these a synthesis of the following equations:\n",
+    "$$\n",
+    "\n",
+    "\\begin{align*}\n",
+    "\\dot{x} &= Ax + Bu & \\qquad u &= -K\\hat{x} \\\\\n",
+    "\\dot{\\hat{x}} &= A\\hat{x} + Bu + L(y-\\hat{y}) & \\qquad y &= Cx \\\\\n",
+    " & & \\qquad \\hat{y} &= C\\hat{x} \\\\\n",
+    "\\end{align*}\n",
+    "\n",
+    "$$\n",
+    "\n",
+    "The last step in constructing this LQG system is to say that the closed loop controller has a nominal scalar gain, $m$. That brings our full system matrix to:\n",
+    "$$\n",
+    "\\begin{bmatrix}\n",
+    "A & mBK\\\\\n",
+    "LC & A-LC-BK \n",
+    "\\end{bmatrix} =\n",
+    "\\begin{bmatrix}\n",
+    "1 & 1 & 0 & 0\\\\\n",
+    "0 & 1 & -mf & -mf\\\\\n",
+    "d & 1 & 1-d & 1\\\\\n",
+    "d & 0 & -d-f & 1-f\\\\\n",
+    "\\end{bmatrix} \n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Exposing Instability\n",
```

Comment:
There are a few sections where the equation gets cut off of the page. Such as the characteristic polynomial in the exposing instability section and the equations after. I suggest trying display the equation on multiple lines. The formatting on the two coefficient terms is a little confusing at first glance as well. It looked like a mistake until I realized that the inequalities were aligned. There are extra $$ characters.

Reply:
I have cleaned up the long polynomial by splitting it into multiple lines.  And I removed the extra dollar signs (just a formatting issue) from the inequalities.  Thanks.